cmd/test/cmd.go

```diff
@@ -0,0 +1,251 @@
+		if err != nil {
+			return err
+		}
+		options.ManifestPath = uptManifestPath
+	}
+
+	manifest, err := model.GetManifestV2(options.ManifestPath, fs)
+	if err != nil {
+		return err
+	}
+
+	k8sClientProvider := okteto.NewK8sClientProviderWithLogger(k8sLogger)
+
+	pc, err := pipelineCMD.NewCommand()
+	if err != nil {
+		return fmt.Errorf("could not create pipeline command: %w", err)
+	}
+
+	configmapHandler := deployCMD.NewConfigmapHandler(k8sClientProvider, k8sLogger)
+
+	builder := buildv2.NewBuilderFromScratch(at, ioCtrl)
+
+	c := deployCMD.Command{
+		GetManifest: func(path string, fs afero.Fs) (*model.Manifest, error) {
+			return manifest, nil
+		},
+		K8sClientProvider:  k8sClientProvider,
+		Builder:            builder,
+		GetDeployer:        deployCMD.GetDeployer,
+		EndpointGetter:     deployCMD.NewEndpointGetter,
+		DeployWaiter:       deployCMD.NewDeployWaiter(k8sClientProvider, k8sLogger),
+		CfgMapHandler:      configmapHandler,
+		Fs:                 fs,
+		PipelineCMD:        pc,
+		AnalyticsTracker:   at,
+		IoCtrl:             ioCtrl,
+		K8sLogger:          k8sLogger,
+		IsRemote:           env.LoadBoolean(constants.OktetoDeployRemote),
+		RunningInInstaller: config.RunningInInstaller(),
+	}
+	if err = c.Run(ctx, &deployCMD.Options{
+		Manifest:         manifest,
+		ManifestPathFlag: options.ManifestPathFlag,
+		ManifestPath:     options.ManifestPath,
+		Name:             options.DeployName,
+		Namespace:        options.Namespace,
+		K8sContext:       options.K8sContext,
+		ServicesToDeploy: options.DeployServices,
+		Variables:        options.Variables,
+		Build:            options.Build,
+		Dependencies:     options.Dependencies,
+		Timeout:          options.Timeout,
+		RunWithoutBash:   options.DeployWithoutBash,
+		RunInRemote:      options.DeployInRemote,
+		Wait:             options.Wait,
+		ShowCTA:          false,
+	}); err != nil {
+		oktetoLog.Error("deploy failed: ", err.Error())
+		return err
+	}
+
+	name := "unit"
+
+	commandsFlags, err := deployCMD.GetCommandFlags(name, options.Variables)
+	if err != nil {
+		return err
+	}
+
+	runner := remote.NewRunner(ioCtrl, buildCMD.NewOktetoBuilder(
+		&okteto.ContextStateless{
+			Store: okteto.GetContextStore(),
+		},
+		fs,
+	))
+
+	test := manifest.Test["unit"]
+
+	commands := make([]model.DeployCommand, len(test.Command))
+
+	for i, cmd := range test.Command {
+		commands[i] = model.DeployCommand{
+			Name:    fmt.Sprintf("command %v", i+1),
+			Command: cmd,
+		}
+	}
+
+	return runner.Run(ctx, &remote.Params{
+		BaseImage:           test.Image,
+		ManifestPathFlag:    options.ManifestPathFlag,
+		TemplateName:        "dockerfile",
+		CommandFlags:        commandsFlags,
+		BuildEnvVars:        builder.GetBuildEnvVars(),
+		DependenciesEnvVars: deployCMD.GetDependencyEnvVars(os.Environ),
+		DockerfileName:      "Dockerfile.test",
+		Deployable: deployable.Entity{
+			Commands: commands,
+		},
+		Manifest: manifest,
+		Command:  remote.DeployCommand, // TODO: use test?
```

Comment:
As discussed in the daily, maybe it worths to create a specific command for test. The `remote-run deploy` does more things than just execute the list of commands:
* It starts the local proxy to forward all kubernetes calls
* It deploys external resources
* It deploys divert information 

So, it might print misleading logs in that regard or have a side effect

In fact, `destroy` is much more similar to `test` than `deploy`. We might even change the name of the remote commands to reuse destroy and test. But given pst experiences, I would have them separated (having main common logic)

Reply:
I'll do this in a follow-up. To avoid this PR getting too big.